### ISSUE TITLE
Feature to clean/purge nodes from a pool

### ISSUE THREAD

**Issue by jvogt** [target/f5-bigip-cookbook#14]
If I understand the readme, setting the members property of f5_ltm_pool will only add members.

I'd like to be able to _set_ the members using this prop.

Proposing adding an additional bool, _purge_, defaulting to false, which would remove any nodes not in the pool.

Additionally, I'd like the option to delete the node record itself, but I'm not sure if that should just be another property (delete_node_during_purge?) or if there's a better idea.

Wanted to invite discussion before I work towards a solution.

**Comment by monkey1016** [comment]
@jvogt I created a pull request that might address what you're looking for.  Can you try it out and let me know how it works?

**Comment by jimlitton** [comment]
Can we just have an update option?  If I understand the new code monkey1016 is working on, I can't just do one call to update a pool, add and remove members?  Will I need to call create and then clear?   I  tested effectively make one call changing the create to a createOrUpdate . . .

Looking at the provider_ltm_pool.rb I found by inserting a “remove_member_v2” call in the action_create function I could remove unused  members.

  def action_create # rubocop:disable AbcSize
        create_pool unless current_resource.exists
        set_lb_method unless current_resource.lb_method == new_resource.lb_method
        load_balancer.client['LocalLB.Pool'].remove_member_v2([new_resource.pool_name], [remove_members])
    . . . 
        
I inserted a remove_members definition after the def missing_members -- I just added "unless current_members.empty?" after seeing money1016's code.  I'm a whole 5+ days into chef & ruby now I think . . .

  def remove_members
    remove_members = current_members - new_members unless current_members.empty?
  end
